### ISSUE TITLE
Packaging KA Lite as a native Chrome app (for ChromeOS and beyond!)

### ISSUE THREAD

**Issue by jamalex** [learningequality/ka-lite#3694]
We dreamed of this day. Well, *that* day, to come. Now, all we need is a hero!

At our official KA Lite launch event at Khan Academy in February of 2013, Jaime Casap's first question was "So when can we get this running on Chromebooks?". At the time, we didn't have an answer.

It turns out that NaCl supports Python, and [there's a Chrome App for that](https://chrome.google.com/webstore/detail/python/nodpmmidbgeganfponihbgmfcoiibffi/related?hl=en). And, as of recently, it supports sockets (`import socket`). So, all the pieces are now in place for a KA Lite Chrome App, which will enable standalone use on a Chromebook, and easy installation on any computer running Chrome as well.

Some good discussion here (including how to build a Chrome app bundling Python and whatever other code you have, and wsgi support etc):
https://groups.google.com/forum/#!topic/native-client-discuss/ODrSbdJoPdo

Let's make it happen! Who's up for giving this a shot?

**Comment by jamalex** [comment]
https://chromium.googlesource.com/external/naclports/+/master/ports/python

**Comment by eff-kay** [comment]
I ll have a look into it , but I dont give you any gurantees cause I have a full time job you know....

**Comment by eff-kay** [comment]
Has anything been done on this before, any discussions or brainstorming that might come in helpful...

**Comment by aronasorman** [comment]
Hi @eff-kay, we haven't had any in-depth discussion how to do it yet, but @benjaoming has been working on installing KA Lite into a python system before.

Once you have python working in your system you can run `python setup.py install` inside `ka-lite` to install it to your python's `site-packages`. Once there you should have all the dependencies needed to run KA Lite. Ping us on hipchat if you have trouble getting to this point, or even once you've successfully gotten to this point, so we can discuss it further!

**Comment by jamalex** [comment]
@aronasorman I think @eff-kay is asking whether there's been any other discussion of how to get this running in a Chrome app, not on a python system?

@eff-kay: Thanks for jumping in! The Google Group link in the description of this issue above shows a discussion of how to build a Chrome app with Python itself bundled in along with a Python application. Probably starting out with getting the standard vanilla Python Chrome app building (or an example bundled app), then get the KA Lite source code in there (working from the develop branch) and have it launch "bin/kalite start". Let us know how it goes!

**Comment by aronasorman** [comment]
@jamalex yup, got that. Was just giving hints since one thing we DO know about the system is that it has Python. However, one thing I'm not sure if a Chrome app has access to is a plain bash shell, so I'm not sure if just running `kalite start` would work.

**Comment by eff-kay** [comment]
@jamalex thnx for the info will look into it...

**Comment by willmee** [comment]
Curious if this is still actively being pursued? It would be very cool to have KALite installed on a ChromeBook.
